### DISCUSSION TITLE
WDIO v6 - browser.mock() does not work if session is reloaded

### DISCUSSION THREAD

**Discussion by johnson-tyl** [webdriverio/webdriverio#6688] [Q&A]
We are using `browser.mock()` to spy on non XHR calls and we have situation where we need to do a `browser.sessionReload()` which causes browser.mock() to stop working on any new window opened. Is this is expected? And is there a workaround to this? We cant use `clearCookies()` as we have multiple applications and the clearingCookies at the end only clears cookies of the last opened domain.

Any help on this is appreciated.

**Comment by christian-bromann** [comment]
@johnson-tyl what WebdriverIO version are you using?

**Reply by johnson-tyl** [reply]
V6

**Reply by christian-bromann** [reply]
We recommend to update to v7 where this bug has been fixed.

**Reply by johnson-tyl** [reply]
@christian-bromann Thanks. Can we upgrade to v7 without breaking the current tests, please?

**Reply by christian-bromann** [reply]
yes

**Reply by johnson-tyl** [reply]
Hi @christian-bromann we just tried it on V7 of WDIO (all libraries upgraded) an still see issues, this is the error message we see:

![image](https://user-images.githubusercontent.com/63800148/116687490-2e877e00-a9ad-11eb-928e-4edee27f020b.png)

**Reply by christian-bromann** [reply]
Seems that you've closed a browser window somewhere.

**Reply by johnson-tyl** [reply]
Yes, `browser.reloadSession()` closes and reopens the browser, that's what the original problem was in v6. Calling `browser.mock()` works for the first test, in `afterScenario` we run `browser.reloadSession()` and when the next test starts and if there's a `browser.mock()`, then this error is seen.

We have upgraded everything WDIO module to latest v7.

Any advice, please?

**Reply by christian-bromann** [reply]
Can you raise an issue for this please?

**Reply by johnson-tyl** [reply]
Sure, will log an issue. Thanks

**Reply by johnson-tyl** [reply]
Done, https://github.com/webdriverio/webdriverio/issues/6786
